packages/core/core/src/BundleGraph.js

```diff
@@ -1458,7 +1450,9 @@ export default class BundleGraph {
 
   getHash(bundle: Bundle): string {
     let hash = new Hash();
-    hash.writeString(bundle.id + this.getContentHash(bundle));
+    hash.writeString(
+      bundle.id + bundle.target.publicUrl + this.getContentHash(bundle),
```

Comment:
Target options aren't in the environment hash and aren't tracked in the graph. This is kinda assuming that packagers will use the publicUrl in some way, but currently this is only true of the HTML packager. Perhaps we can make this more granular in the future somehow...